tests/TestMembershipContextFactory.php

```diff
@@ -6,28 +6,30 @@
 
 use Doctrine\DBAL\Connection;
 use Doctrine\DBAL\DriverManager;
+use Doctrine\DBAL\Tools\DsnParser;
 use Doctrine\ORM\EntityManager;
 use Doctrine\ORM\ORMSetup;
 use WMDE\Fundraising\MembershipContext\MembershipContextFactory;
 use WMDE\Fundraising\PaymentContext\PaymentContextFactory;
 
 class TestMembershipContextFactory {
 
-	private array $config;
 	private ?Connection $connection;
 	private MembershipContextFactory $factory;
 	private ?EntityManager $entityManager;
 
-	public function __construct( array $config ) {
-		$this->config = $config;
+	public function __construct() {
 		$this->factory = new MembershipContextFactory();
 		$this->entityManager = null;
 		$this->connection = null;
 	}
 
 	public function getConnection(): Connection {
 		if ( $this->connection === null ) {
-			$this->connection = DriverManager::getConnection( $this->config['db'] );
+			$dsnParser  = new DsnParser( [ 'sqlite' => 'pdo_sqlite' ] );
+			$this->connection = DriverManager::getConnection(
+				$dsnParser->parse( 'sqlite:///:memory:' )
+			);
```

Comment:
This hard-codes sqlite as our test database. Instead of doing that, please add a `$dbConfig` parameter for the `TestMembershipContextFactory` and type it with the `Param` type that you can import it from `DriverManager`, see  https://github.com/doctrine/dbal/blob/9e588fe1f38a443cb17de6b86b803d9e028e2156/src/DriverManager.php#L46-L68
I've encountered this problem before and wrote a blog article about it: https://lebenplusplus.de/2022/05/05/using-doctrine-connection-array-shapes-with-phpstan/

Reply:
Done.